src/formatDate.js

```diff
@@ -7,8 +7,50 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(date, sourceFormat, targetFormat) {
+  // Делаем деструктуризацию кода
+  const [, , , sourceSeparator] = sourceFormat;
+  const [, , , targetSeparator] = targetFormat;
+
+  // Разделение даты по сепаратору
+  const dateParts = date.split(sourceSeparator);
+  const [sourceYear, sourceMonth, sourceDay] = dateParts;
+
+  const formattedDateParts = [];
+
+  // Переформатирование даты в зависимости от исходного формата
+  if (
+    sourceFormat[0] === 'YYYY' &&
+    targetFormat[0] === 'DD' &&
+    targetFormat[1] === 'MM' &&
+    targetFormat[2] === 'YYYY'
+  ) {
+    formattedDateParts.push(sourceDay, sourceMonth, sourceYear);
+  } else if (targetFormat[2] === 'YY' && sourceFormat[2] === 'YYYY') {
+    formattedDateParts.push(
+      dateParts[0],
+      dateParts[1],
+      dateParts[2].slice(2, 4),
+    );
+  } else if (sourceFormat[0] === 'YY' && targetFormat[0] === 'YYYY') {
+    const fullYear = sourceYear < 30 ? `20${sourceYear}` : `19${sourceYear}`;
+
+    formattedDateParts.push(fullYear, sourceMonth, sourceDay);
+  } else if (sourceFormat[0] === 'YY' && targetFormat[0] === 'DD') {
+    const fullYear = sourceYear < 30 ? `20${sourceYear}` : `19${sourceYear}`;
+
+    formattedDateParts.push(sourceDay, sourceMonth, fullYear);
+  } else if (
+    sourceFormat[0] === targetFormat[0] &&
+    sourceFormat[1] === targetFormat[1] &&
+    sourceFormat[2] === targetFormat[2]
+  ) {
+    return dateParts.join(targetSeparator);
+  } else {
+    formattedDateParts.push(dateParts[2], dateParts[0], dateParts[1]);
```

Comment:
The 'else' block assumes a specific order of date parts. It is better to write a more dynamic solution that does not rely on the date parts being in a specific order. You should iterate over the targetFormat and construct the formattedDateParts based on the order specified in the targetFormat.